lib/cocoapods/installer/analyzer/target_inspector.rb

```diff
@@ -159,6 +159,10 @@ def compute_platform(user_targets)
                   "Unable to determine the platform for the `#{target_definition.name}` target."
           end
 
+          UI.warn "Automatically assigning platform #{name} with version #{deployment_target} " \
```

Comment:
I think this is a good change by the way. I had no idea cocoapods infers the platform automatically. Given most users include a `platform` directive I think this provides good visibility.